src/plugins/workspace/public/hooks.ts

```diff
@@ -3,12 +3,13 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { ApplicationStart, PublicAppInfo } from 'opensearch-dashboards/public';
 import { useObservable } from 'react-use';
 import { useMemo } from 'react';
+import { of } from 'rxjs';
+import { ApplicationStart, PublicAppInfo } from '../../../core/public';
 
-export function useApplications(application: ApplicationStart) {
-  const applications = useObservable(application.applications$);
+export function useApplications(application?: ApplicationStart) {
+  const applications = useObservable(application?.applications$ ?? of(new Map()), new Map());
   return useMemo(() => {
     const apps: PublicAppInfo[] = [];
     applications?.forEach((app) => {
```

Comment:
May not related to this PR but why don't we use `.map` here?

Reply:
Yes, this is added by other PRs, looks like map is more appropriate, I will update it.

Reply:
The type of applications here is `ReadonlyMap`, so it doesn't support map.

Reply:
Thanks for the clarification.